tests/src/test/java/alluxio/server/configuration/ConfigCheckerIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+        .setNumWorkers(TEST_NUM_WORKERS)
+        .setWorkerProperties(workerProperties)
+        .build();
+
+    try {
+      cluster.start();
+      MetaMasterClient client = cluster
+          .waitForAllNodesRegistered(30 * Constants.SECOND_MS);
+
+      Assert.assertEquals(1, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_MASTER_NUM)))
+          .getConfMasterNum());
+      Assert.assertEquals(TEST_NUM_WORKERS, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_WORKER_NUM)))
+          .getConfWorkerNum());
+      cluster.notifySuccess();
+    } finally {
+      cluster.destroy();
+    }
+  }
+
+  @Test
+  public void MultiNodes() throws Exception {
+    // Prepare properties
+    Map<Integer, Map<PropertyKey, String>> properties = generateProperties(
+        TEST_NUM_MASTERS + TEST_NUM_WORKERS, PropertyKey.Scope.SERVER);
+    Map<Integer, Map<PropertyKey, String>> masterProperties = properties.entrySet().stream()
+        .filter(entry -> (entry.getKey() < TEST_NUM_MASTERS))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    Map<Integer, Map<PropertyKey, String>> workerProperties = properties.entrySet().stream()
+        .filter(entry -> (entry.getKey() >= TEST_NUM_MASTERS))
+        .collect(Collectors.toMap(entry -> entry.getKey() - TEST_NUM_MASTERS, Map.Entry::getValue));
+
+    MultiProcessCluster cluster = MultiProcessCluster.newBuilder()
+        .setClusterName("ConfigCheckerMultiNodesTest")
+        .setNumMasters(TEST_NUM_MASTERS)
+        .setNumWorkers(TEST_NUM_WORKERS)
+        .setDeployMode(MultiProcessCluster.DeployMode.ZOOKEEPER_HA)
+        .setMasterProperties(masterProperties)
+        .setWorkerProperties(workerProperties)
+        .build();
+
+    try {
+      cluster.start();
+      MetaMasterClient client = cluster.waitForAllNodesRegistered(
+          30 * Constants.SECOND_MS);
+
+      Assert.assertEquals(TEST_NUM_MASTERS, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_MASTER_NUM)))
+          .getConfMasterNum());
+      Assert.assertEquals(TEST_NUM_WORKERS, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_WORKER_NUM)))
+          .getConfWorkerNum());
+      cluster.notifySuccess();
+    } finally {
+      cluster.destroy();
+    }
+  }
+
+  /**
+   * Generates a map that different nodes contain different values for one property.
+   *
+   * @param nodeNum the number of nodes to test
+   * @param scope the scope of the property key
+   * @return generated properties
+   */
+  private Map<Integer, Map<PropertyKey, String>> generateProperties(int nodeNum,
+      Scope scope) {
+    Map<Integer, Map<PropertyKey, String>> properties = new HashMap<>();
+    // Choose some example properties
+    PropertyKey key = scope.equals(Scope.MASTER) ? PropertyKey.MASTER_JOURNAL_FLUSH_TIMEOUT_MS
```

Comment:
could we pass in the property instead of the scope? we only use the scope to figure out the property

Reply:
Changed from passing scope to passing property key